frontend/src/components/question-service/edit-question-page/EditQuestionForm.tsx

```diff
@@ -0,0 +1,128 @@
+import { EMPTY_QUESTION, Question } from "@/api/question-service/Question";
+import { fetchQuestionById, insertQuestion, updateQuestion } from "@/api/question-service/QuestionService";
+import { useEffect, useState } from "react";
+import PageTitle from "../../common/PageTitle";
+import QuestionInputField from "./QuestionInputField";
+import { Button } from "../../ui/button";
+import { Link, useNavigate } from "react-router-dom";
+import QuestionTextareaField from "./QuestionTextareaField";
+import QuestionSelectField from "./QuestionSelectField";
+import { TDifficulty } from "../Difficulty";
+import QuestionTopicsField from "./QuestionTopicsField";
+import { TriangleAlert } from "lucide-react";
+
+/**
+ * A form for the editing of questions. Requires the following
+ * prop:
+ * 
+ * - `id`: The ID of the question to edit. Set `id` to `undefined` to create
+ * a new question instead of updating an existing question.
+ */
+export default function EditQuestionForm({ id } : { id? : string }) {
+  const [ question, setQuestion ] = useState<Question>(EMPTY_QUESTION);
+  const navigate = useNavigate();
+
+  // whether to perform insert or update. True if adding question, False if
+  // updating an existing question.
+  const isAddQuestion = (id === undefined);
+
+  // load question from backend
+  const loadQuestion = () => {
+    fetchQuestionById( id ).then(question => {
+      setQuestion(question);
+      console.log(question);
+    });
+  }
+
+  useEffect(() => {
+    loadQuestion();
+  }, [])
+
+  const [ warning, setWarning ] = useState("");
+
+  /**
+   * Handles the create/update of the question.
+   */
+  const handleSaveQuestionClick = () => {
+    if (question.title === "" || question.description === "") {
+      // frontend validation: title and description missing
+      setWarning("Title and description are required!");
+    } else if (isAddQuestion) {
+      // adds a question to the database
+      insertQuestion(question).then(response => {
+        if (response.status === 200) {
+          navigate("/questions");
+        } else {
+          setWarning(response.message as string);
+        }
+      });
+    } else {
+      // updates the question in the database
+      updateQuestion(question).then(response => {
+        if (response.status === 200) {
+          navigate("/questions");
+        } else {
+          setWarning(response.message as string);
+        }
+      });
+    }
+  }
+
+  return (
+    <section>
+      <PageTitle>
+        { isAddQuestion ?
+          "Add New Question" :
+          <>Edit Question #{id}</>
```

Comment:
It looks fine here but the ss shows that both pages has the edit question title

Reply:
the ss is old before this update. updated the ss